### PR TITLE
waagent: update to 2.8.0.11

### PR DESCRIPTION
**Pull request by asmundo**
The agent has no been updated for a very long time. In addition to updating to a newest tagged version the waagent it is moved in to a package.

The existing version has issues with the new python2.7 package not containing crypt.so file. And the commit
6910a4eea0038728a2f10ce84122806f2cb6b170 I believe introduced regression that caused the shebang to not be updated.

###### Description of changes
 
Last time this package was updated was 7 years ago. Changelog do not make that much sense in this case. But:
https://github.com/Azure/WALinuxAgent/releases/tag/v2.8.0.11

I am still new to nix and happy to improve my pull request.  

###### Things done

Made a pkg of the agent and updated it to latest released.  

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by zimbatm** [comment]
Result of `nixpkgs-review pr 206974` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
    <li>tests.nixos-functions.nixos-test</li>
  </ul>
</details>
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>waagent</li>
  </ul>
</details>

**Comment by ElXreno** [comment]
You should also update the following lines:
```diff
diff --git a/nixos/modules/virtualisation/azure-agent.nix b/nixos/modules/virtualisation/azure-agent.nix
index 21f69d86d87..6e6021cf80f 100644
--- a/nixos/modules/virtualisation/azure-agent.nix
+++ b/nixos/modules/virtualisation/azure-agent.nix
@@ -200,7 +200,7 @@ in
         OS.MonitorDhcpClientRestartPeriod=30
     '';
 
-    services.udev.packages = [ waagent ];
+    services.udev.packages = [ pkgs.waagent ];
 
     networking.dhcpcd.persistent = true;
 
@@ -245,7 +245,7 @@ in
       description = "Windows Azure Agent Service";
       unitConfig.ConditionPathExists = "/etc/waagent.conf";
       serviceConfig = {
-        ExecStart = "${waagent}/bin/waagent -daemon";
+        ExecStart = "${pkgs.waagent}/bin/waagent -daemon";
         Type = "simple";
       };
     };
```

**Comment by asmundo** [comment]
Thanks @ElXreno for catching this! I have not found a way of testing/using local version of nixpkgs. :-/

I am used to Gerrit that so I did a fetch and rebase on top of master before pushing updated version of the commit. So finding out if I did fixed it in a glans is not that easy. But you can look at:
https://github.com/NixOS/nixpkgs/compare/da383d2f5ddf3a0702dc1d3cc291508f8aa4c957..e3a248ac9f97ccf193c2d9aa801aed26c8ab52c3#diff-12ee33e8774acef5711615d1e4cfab71409b7710cd776c6f95db2991394b4c06

@zimbatm Thank you for giving feedback. But I do not understand if you comment is a 👍🏿 or a 👎🏿  and requires actions from me. Please let me know.

**Comment by ElXreno** [comment]
> I have not found a way of testing/using local version of nixpkgs

I'm using `azure-common` virtualization module for my server, you can check it here: https://github.com/ElXreno/nixos-config/blob/fca628f208a7240647c1e7d0b0973cc62098a730/machines/Nixis-Server/default.nix#L7
I checked the changes by updating my server with this PR.

**Review by zimbatm — APPROVED** [review]
The change looks good, but I don't have a VM to test it. I'm going to defer to you both and assume that it's tested.

File: nixos/modules/virtualisation/azure-agent.nix [hunk 1]
```diff
@@ -1,51 +1,10 @@
...
-    pythonPath = [ pythonPackages.pyasn1 ];
-
-    configurePhase = false;
-    buildPhase = false;
-
-    installPhase = ''
-      substituteInPlace config/99-azure-product-uuid.rules \
-          --replace /bin/chmod "${coreutils}/bin/chmod"
-      mkdir -p $out/lib/udev/rules.d
-      cp config/*.rules $out/lib/udev/rules.d
```
Comment (Atry) [inline nixos/modules/virtualisation/azure-agent.nix:37]: Why did you remove these code?

Comment (Atry) [reply nixos/modules/virtualisation/azure-agent.nix:37]: @asmundo I believe currently the `udev` rules do not take effect because of this PR. See https://github.com/NixOS/nixpkgs/pull/266310#issuecomment-1826419505
